server/app/views/admin/questions/QuestionEditView.java

```diff
@@ -457,10 +468,94 @@ private DomContent buildUniversalQuestion(QuestionForm questionForm) {
                 /* text= */ Optional.of("Set as a universal question")));
   }
 
+  private FieldsetTag buildPrimaryApplicantInfoSection(QuestionForm questionForm) {
+    FieldsetTag result =
+        fieldset()
+            .withId("primary-applicant-info")
+            .with(legend("Primary Applicant Information").withClass(BaseStyles.INPUT_LABEL));
+    PrimaryApplicantInfoTag.getAllTagsForQuestionType(questionForm.getQuestionType())
+        .forEach(
+            primaryApplicantInfoTag -> {
+              Optional<QuestionDefinition> currentQuestionForTag =
+                  questionService.getReadOnlyQuestionServiceSync().getUpToDateQuestions().stream()
+                      .filter(
+                          qd -> qd.getPrimaryApplicantInfoTags().contains(primaryApplicantInfoTag))
+                      .findFirst();
+              boolean differentQuestionHasTag =
+                  currentQuestionForTag
+                      .map(question -> !question.getName().equals(questionForm.getQuestionName()))
+                      .orElse(false);
+              DivTag tagSubsection = div().withClass("cf-primary-applicant-info-subsection");
+
+              if (differentQuestionHasTag) {
+                // Only include the alert. We'll swap the text based on the state of the
+                // universal toggle in admin_question_edit.ts.
+                String otherQuestionName =
+                    currentQuestionForTag.map(QuestionDefinition::getName).orElseThrow();
+                String alreadySetAlertText =
+                    String.format(
+                        "You cannot edit this setting since this property is already set on a"
+                            + " question named %s.",
+                        otherQuestionName);
+                String nonUniversalAlreadySetAlertText =
+                    String.format(
+                        "You cannot edit this setting since the question is not a universal"
+                            + " question and because this property is already set on a question"
+                            + " named %s.",
+                        otherQuestionName);
+                String initialAlertText =
+                    questionForm.isUniversal()
+                        ? alreadySetAlertText
+                        : nonUniversalAlreadySetAlertText;
+                tagSubsection =
+                    tagSubsection
+                        .with(
+                            ViewUtils.makeAlertInfoSlim(
+                                initialAlertText,
+                                /* hidden= */ false,
+                                /* classes...= */ "cf-primary-applicant-info-alert",
+                                "usa-alert-primary-applicant-info"))
+                        .withData("already-set-alert", alreadySetAlertText)
+                        .withData(
+                            "non-universal-already-set-alert", nonUniversalAlreadySetAlertText);
```

Comment:
*suggestion (blocking but I could be convinced otherwise)*: I think this view code, and the JS that manipulates the DOM it renders, would be simpler if we put everything we might need on the page, but hidden.

Instead of stashing the error messages we might need in a `data` attribute and pulling them out in the JS, I think we should include a few versions of the alert in the DOM and unhide the one that applies.

That simplifies this class because then we can structure it like
```java
DivTag alertNotUniversalSection = ...
DivTag alertNotUniversalAndAlreadySetSection = ...
DivTag alertAlreadySetSection = ...

tagSubsection
  .condWith(alertNotUniversal, alertNotUniversalSection)
  .condWith(alertNotUniversalAndAlreadySet, alertNotUniversalAndAlreadySetSection)
  .condWith(alertAlreadySet, alertAlreadySetSection)
  .condWith(includeToggle, toggleButton);
```

And if we're ok having two alert boxes, we don't need the middle permutation.

Reply:
I found that easier to reason about! I get the page-weight concern, but I think it'll mostly be gzipped out, and I don't have concerns about hiding stuff in the dom until we need it. (that's what we do with modals, after all)

Reply:
Is the "may be null" comment because the element might not be on the page if we're in the "it's already set on another question" condition?